starport/services/scaffolder/module.go

```diff
@@ -212,6 +224,36 @@ func moduleExists(appPath string, moduleName string) (bool, error) {
 	return true, err
 }
 
+func checkModuleName(moduleName string) error {
+	// go keyword
+	if token.Lookup(moduleName).IsKeyword() {
+		return fmt.Errorf("%s is a Go keyword", moduleName)
+	}
+
+	// name of default registered module
+	switch moduleName {
+	case
+		"auth",
```

Comment:
can we dynamically get this list with code analysis?

first we need to get src path of sdk with this: https://github.com/tendermint/starport/blob/develop/starport/pkg/gomodule/gomodule.go#L76

then we need to `ls x/` to get this list.

Reply:
This is a little bit more complicated because not all modules are in `cosmos-sdk` like ibc modules (in the future) and some module names don't match the directory name `distribution` -> `distr`, ibc-transfer -> `transfer`. IMO this is simpler to maintain a list it should not be updated frequently in the future